api/config/cdm/cdm_5_2.json

```diff
@@ -0,0 +1,1269 @@
+{
+  "description": "Configuration specifying the schema for tables in the CDR, including foreign key relationships. Adapted from BigQuery schema files, found in https://github.com/all-of-us/curation/data_steward/resources/fields.",
+  "version": "5.2",
+  "metadata_tables": {
+    "care_site": {
+      "columns": [
+        {
+          "type": "integer",
+          "name": "care_site_id",
+          "mode": "required",
+          "description": "A unique identifier for each Care Site.",
+          "primaryKey": true
+        },
+        {
+          "type": "string",
+          "name": "care_site_name",
+          "mode": "nullable",
+          "description": "The verbatim description or name of the Care Site as in data source"
+        },
+        {
+          "type": "integer",
+          "name": "place_of_service_concept_id",
+          "mode": "nullable",
+          "description": "A foreign key that refers to a Place of Service Concept ID in the Standardized Vocabularies.",
+          "foreignKey": "concept"
```

Comment:
My understanding is that this is BigQuery schema JSON with the addition of this `foreignKey`, is that right? If so, this sounds reasonable.

Would be good to check in the script used to generate this file, to also serve as documentation of where this came from.

Would we associate "cdm_5_2" with a particular CDR version, in order to know which JSON to use here?

Reply:
Yep, that and primaryKey. I can imagine it expanding in future to include other things, but for now those are the only changes.

There wasn't a script; I made this file by copying the contents of the files in curation/data_steward/resources/files by hand into here. (I figured it would take me more time to write a script, and in future we'll want to update this incrementally anyway.)

Regarding CDR versions: it probably does make sense to tag them with a CDM version so we know which schema to use with them. Though we may wind up upgrading all CDR versions to be the latest CDM schema when we switch to support a new CDM schema... I don't think that has been worked out yet.

Reply:
I would at least document in a README where you got this file and what transformations you made; it seems very likely that if we were to upgrade to a new version of the CDM schema we'd just copy that file in its entirety again and make the same transformations.

Reply:
Done.